### ISSUE TITLE
Wrong implement of formula?

### ISSUE THREAD

**Issue by iron9light** [JosvanderWesthuizen/janet#1]
https://github.com/JosvanderWesthuizen/janet/blob/45b89d4740139066745d19c0e8a70dea67175be9/aux_code/rnn_cells.py#L63

Should be:
```py
f = tf.sigmoid(f)
new_c = f * c + (1 - f) * tf.tanh(j)
```

**Comment by JosvanderWesthuizen** [comment]
Thanks for pointing this out. This is a bias that we found helpful to subtract from the input modulation component. The paper has been updated to mention this requirement.

Essentially sigmoid(1-f) = 1 - sigmoid(f-1).
You can change it to the standard (subtracting zero) with: sigmoid(-f) = 1-sigmoid(f)

In future changes I'll update the code to be more explicit.

**Comment by iron9light** [comment]
Where can I find the updated paper? I cannot find the new version at https://arxiv.org/abs/1804.04849
I'm interested in why it's helpful.
If this trick helps, maybe we can use it for GRU as well.

**Comment by JosvanderWesthuizen** [comment]
@iron9light sorry, I was jumping the gun a bit. The new version is currently being processed on arxiv. It should be public by end of Monday.
Empirically, we found the model to do better when we subtracted 1. It seems sensible to let more information accumulate than the amount forgotten.
The value subtracted could be data dependent, but among all the values tried, 1 provided the best results on the datasets we used.
It's a good idea to see if this is useful for the GRU as well.
